Sources/ScheduledTasks.php

```diff
@@ -1542,4 +1542,196 @@ function scheduled_remove_old_drafts()
 	return true;
 }
 
+/**
+ * Prune log_topics, log_boards & log_mark_boards_read.
+ * For users who haven't been active in a long time, purge these records.
+ * For users who haven't been active in a shorter time, mark boards as read,
+ * pruning log_topics.
+ */
+function scheduled_prune_log_topics()
+{
+	global $smcFunc, $sourcedir, $modSettings;
+
+	if (empty($modSettings['mark_read_max_users']))
+		return true;
+
+	// Convert to timestamps for comparison
+	$markReadCutoff = time() - $modSettings['mark_read_beyond'] * 86400;
+	$cleanupBeyond = time() - $modSettings['mark_read_delete_beyond'] * 86400;
+	$maxMembers = $modSettings['mark_read_max_users'];
+
+	// Try to prevent timeouts
+	@set_time_limit(300);
+	if (function_exists('apache_reset_timeout'))
+		@apache_reset_timeout();
+
+	// Start off by finding the records in log_boards, log_topics & log_mark_read
+	// for users who haven't been around the longest...
+
+	// Find the oldest $maxMembers in log_boards
+	$sql = "SELECT lb.id_member, m.last_login
+		FROM {db_prefix}members m
+		INNER JOIN
+		(
+			SELECT DISTINCT id_member
+			FROM {db_prefix}log_boards
+		) lb
+		ON lb.id_member = m.id_member
+		WHERE m.last_login <= {int:cutoff}
+		ORDER BY m.last_login
+		LIMIT {int:limit}";
+	$result = $smcFunc['db_query']('', $sql,
+		array(
+			'limit' => $maxMembers,
+			'cutoff' => $cleanupBeyond,
+		)
+	);
+	$members = array();
+	while($row = $smcFunc['db_fetch_assoc']($result))
+		$members[$row['id_member']] = $row['last_login'];
+	$smcFunc['db_free_result']($result);
+
+	// Find the oldest $maxMembers in log_mark_read
+	$sql = "SELECT lmr.id_member, m.last_login
+		FROM {db_prefix}members m
+		INNER JOIN
+		(
+			SELECT DISTINCT id_member
+			FROM {db_prefix}log_mark_read
+		) lmr
+		ON lmr.id_member = m.id_member
+		WHERE m.last_login <= {int:cutoff}
+		ORDER BY m.last_login
+		LIMIT {int:limit}";
+	$result = $smcFunc['db_query']('', $sql,
+		array(
+			'limit' => $maxMembers,
+			'cutoff' => $cleanupBeyond,
+		)
+	);
+	// Using id_member as key basically eats dupes...
+	while($row = $smcFunc['db_fetch_assoc']($result))
+		$members[$row['id_member']] = $row['last_login'];
+	$smcFunc['db_free_result']($result);
+
+	// Find the oldest $maxMembers in log_topics
+	// Note the different cutoff, as we may need to mark these as read
+	$sql = "SELECT lt.id_member, m.last_login
+		FROM {db_prefix}members m
+		INNER JOIN
+		(
+			SELECT DISTINCT id_member
+			FROM {db_prefix}log_topics
+			WHERE unwatched = {int:unwatched}
+		) lt
+		ON lt.id_member = m.id_member
```

Comment:
I would move the on part on tab to the right,
since it's part of the join statement or move a line up with the alias.

Reply:
Will do...